### PR TITLE
Features/python/generic types

### PR DESCRIPTION
**Pull request by Sahloul**
Removes old binders and utilize the generic ones introduced in PR #8319.
The new binders' behavior is guaranteed to be backward-compatible based on the provided tests.

### This pullrequest changes
- Removes old bindings
- Tests the behavior for backward-compatibility

### What is the problem with the in-use bindings?
- Only some vector are supported
- They are implemented in different ways, hence non-uniform behavior
- They accept only tuples (hard-coded)! Numpy arrays are rejected

### CODE REVIEW

**Comment by alalek** [comment]
Please add tests for this.
It seems new version of these conversions are not equal to previous.

File: modules/python/src2/cv2.cpp [hunk 1]
```diff
@@ -933,27 +954,6 @@ PyObject* pyopencv_from(const Point3f& p)
...
-    (void)name;
-    if(!obj)
-        return true;
-    return PyArg_ParseTuple(obj, "ddd", &v[0], &v[1], &v[2]) > 0;
-}
-
-template<>
-PyObject* pyopencv_from(const Vec3d& v)
-{
-    return Py_BuildValue("(ddd)", v[0], v[1], v[2]);
```
Comment (Sahloul) [inline modules/python/src2/cv2.cpp]: @alalek First of all, let me explain what I have done. The removed ones have the format (ddd) or (dd) which is converted into Python [tuples](https://www.tutorialspoint.com/python/python_tuples.htm). However, the binders I have provided converts the data into Numpy arrays which is more flexible and powerful.

Second, although I would like to write some tests to prove that existing Python scripts would not complain from the change, I do not know a good way to show such compatibility. Could you point me out to something related? Or should we keep the old ones and have different behaviors for different vectors?

Comment (Sahloul) [inline modules/python/src2/cv2.cpp]: BTW: I have already tested various examples in the `samples/native/python` directory, and they are doing just fine!

**Comment by Sahloul** [comment]
@alalek I am giving up on convincing you that the new versions are backward-compatible!
Allow me just to push the new types in order to finish the [#1023](https://github.com/opencv/opencv_contrib/issues/1023) cleanup.

**Comment by Sahloul** [comment]
@alalek Mind to check why tests failed?

**Comment by Sahloul** [comment]
@alalek I see your concern, and I agree with you!
So how about wrapping only the missing types, and just grouping the old functions together?
I think raising the question whether we can replace them or not in future releases is enough for now.

**Comment by Sahloul** [comment]
@alalek You got that! I already included a test file and its tests are passing properly.
However, Linux OpenCL is failing due to unrelated issue. Mind to check it?

**Comment by Sahloul** [comment]
@alalek @vpisarev I added some tests for this PR, kindly review!

**Comment by Sahloul** [comment]
@alalek @vpisarev someone, unrelated tests failed! Mind to check?

**Comment by Sahloul** [comment]
@alalek @vpisarev If this is going to take long, then how about merging commit 7d2ca8b for now? That commit does not effect current implementation or removes any binders.

**Comment by vpisarev** [comment]
@Sahloul, sorry, it looks too heavy. Python is not the fastest language in the world, but these new conversions are very, very slow.

**Pull request closed without merging**